diff --git a/dist/cli/cli.js b/dist/cli/cli.js
index f66d9268f20efe54719737fb6bf4fd32cb7a31d8..78e779e313e47c9d1eaa2389d9df76e95def0fa2 100755
--- a/dist/cli/cli.js
+++ b/dist/cli/cli.js
@@ -638,6 +638,8 @@ var runTestImage = ({
         "run",
         "--env",
         `MESSAGE_FILE_PATH=${containerFilePath}`,
+        "--publish",
+        "127.0.0.1:26657:26657",
         "--mount",
         `source=${messageFilePath},target=${containerFilePath},type=bind`,
         ...removeContainerOnExit ? ["--rm"] : [],
