diff --git a/main/generators/customize-utils.js b/main/generators/customize-utils.js
index 39b65d5e80fe28f4295d229f7daa1c6923db8aaf..5553874955b0a4e6b1fb0bf442add6dd425aaf39 100644
--- a/main/generators/customize-utils.js
+++ b/main/generators/customize-utils.js
@@ -32,13 +32,13 @@ const plugin = (builder) => {
         utils_1.UTILS.base64FromBytes = {
             type: 'import',
             path: '@endo/base64',
-            name: 'decodeBase64',
+            name: 'encodeBase64',
             importAs: 'base64FromBytes',
         };
         utils_1.UTILS.bytesFromBase64 = {
             type: 'import',
             path: '@endo/base64',
-            name: 'encodeBase64',
+            name: 'decodeBase64',
             importAs: 'bytesFromBase64',
         };
     }
diff --git a/main/helpers/internal-for-bigint.js b/main/helpers/internal-for-bigint.js
index b90267f01cebfc3f23d4623e7e4065d5d150b08a..0740b1ecd7087f5c87dc8e37f497727e5c6194df 100644
--- a/main/helpers/internal-for-bigint.js
+++ b/main/helpers/internal-for-bigint.js
@@ -3,41 +3,8 @@ Object.defineProperty(exports, "__esModule", { value: true });
 exports.getHelperForBigint = void 0;
 const getHelperForBigint = (options) => {
     return `
-declare var self: any | undefined;
-declare var window: any | undefined;
-declare var global: any | undefined;
-var globalThis: any = (() => {
-  if (typeof globalThis !== 'undefined') return globalThis;
-  if (typeof self !== 'undefined') return self;
-  if (typeof window !== 'undefined') return window;
-  if (typeof global !== 'undefined') return global;
-  throw 'Unable to locate global object';
-})();
-
-const atob: (b64: string) => string =
-  globalThis.atob ||
-  ((b64) => globalThis.Buffer.from(b64, 'base64').toString('binary'));
-
-export function bytesFromBase64(b64: string): Uint8Array {
-  const bin = atob(b64);
-  const arr = new Uint8Array(bin.length);
-  for (let i = 0; i < bin.length; ++i) {
-    arr[i] = bin.charCodeAt(i);
-  }
-  return arr;
-}
-
-const btoa: (bin: string) => string =
-  globalThis.btoa ||
-  ((bin) => globalThis.Buffer.from(bin, 'binary').toString('base64'));
-
-export function base64FromBytes(arr: Uint8Array): string {
-  const bin: string[] = [];
-  arr.forEach((byte) => {
-    bin.push(String.fromCharCode(byte));
-  });
-  return btoa(bin.join(''));
-}
+// PATCH: Use base64Lib: 'endo' instead for bytesFromBase64 and base64FromBytes
+// TODO update Telescope to put these helpers in a separate module so 'helpers' doesn't entail ambient authority
 
 export interface AminoHeight {
   readonly revision_number?: string;
diff --git a/main/helpers/internal.js b/main/helpers/internal.js
index 9df193598d12e194e7e00af400be4994f2235ae2..3e1c3a2a11c8fd4cbda97ab56029f69a794743bd 100644
--- a/main/helpers/internal.js
+++ b/main/helpers/internal.js
@@ -14,39 +14,8 @@ if (_m0.util.Long !== Long) {
 
 export { Long };
 
-declare var self: any | undefined;
-declare var window: any | undefined;
-declare var global: any | undefined;
-var globalThis: any = (() => {
-    if (typeof globalThis !== 'undefined') return globalThis;
-    if (typeof self !== 'undefined') return self;
-    if (typeof window !== 'undefined') return window;
-    if (typeof global !== 'undefined') return global;
-    throw 'Unable to locate global object';
-})();
-
-const atob: (b64: string) => string =
-    globalThis.atob || ((b64) => globalThis.Buffer.from(b64, 'base64').toString('binary'));
-
-export function bytesFromBase64(b64: string): Uint8Array {
-    const bin = atob(b64);
-    const arr = new Uint8Array(bin.length);
-    for (let i = 0; i < bin.length; ++i) {
-        arr[i] = bin.charCodeAt(i);
-    }
-    return arr;
-}
-
-const btoa: (bin: string) => string =
-    globalThis.btoa || ((bin) => globalThis.Buffer.from(bin, 'binary').toString('base64'));
-
-export function base64FromBytes(arr: Uint8Array): string {
-    const bin: string[] = [];
-    arr.forEach((byte) => {
-        bin.push(String.fromCharCode(byte));
-    });
-    return btoa(bin.join(''));
-}
+// PATCH: Use base64Lib: 'endo' instead for bytesFromBase64 and base64FromBytes
+// TODO update Telescope to put these helpers in a separate module so 'helpers' doesn't entail ambient authority
 
 export interface AminoHeight {
     readonly revision_number?: string;
diff --git a/main/helpers/internalForBigInt.js b/main/helpers/internalForBigInt.js
index 07b6e47289a7e96b6835eef4857b18008d267343..6a02c01c0a31bb388ee11a300790ea785e417f03 100644
--- a/main/helpers/internalForBigInt.js
+++ b/main/helpers/internalForBigInt.js
@@ -2,41 +2,8 @@
 Object.defineProperty(exports, "__esModule", { value: true });
 exports.internalForBigInt = void 0;
 exports.internalForBigInt = `
-declare var self: any | undefined;
-declare var window: any | undefined;
-declare var global: any | undefined;
-var globalThis: any = (() => {
-  if (typeof globalThis !== 'undefined') return globalThis;
-  if (typeof self !== 'undefined') return self;
-  if (typeof window !== 'undefined') return window;
-  if (typeof global !== 'undefined') return global;
-  throw 'Unable to locate global object';
-})();
-
-const atob: (b64: string) => string =
-  globalThis.atob ||
-  ((b64) => globalThis.Buffer.from(b64, 'base64').toString('binary'));
-
-export function bytesFromBase64(b64: string): Uint8Array {
-  const bin = atob(b64);
-  const arr = new Uint8Array(bin.length);
-  for (let i = 0; i < bin.length; ++i) {
-    arr[i] = bin.charCodeAt(i);
-  }
-  return arr;
-}
-
-const btoa: (bin: string) => string =
-  globalThis.btoa ||
-  ((bin) => globalThis.Buffer.from(bin, 'binary').toString('base64'));
-
-export function base64FromBytes(arr: Uint8Array): string {
-  const bin: string[] = [];
-  arr.forEach((byte) => {
-    bin.push(String.fromCharCode(byte));
-  });
-  return btoa(bin.join(''));
-}
+// PATCH: Use base64Lib: 'endo' instead for bytesFromBase64 and base64FromBytes
+// TODO update Telescope to put these helpers in a separate module so 'helpers' doesn't entail ambient authority
 
 export interface AminoHeight {
   readonly revision_number?: string;
diff --git a/module/generators/customize-utils.js b/module/generators/customize-utils.js
index d6a50e021e313c26a01efaf26f4d65e332df24ac..92714e18033fc3edf0433660ce511efde2aa283d 100644
--- a/module/generators/customize-utils.js
+++ b/module/generators/customize-utils.js
@@ -29,13 +29,13 @@ export const plugin = (builder) => {
         UTILS.base64FromBytes = {
             type: 'import',
             path: '@endo/base64',
-            name: 'decodeBase64',
+            name: 'encodeBase64',
             importAs: 'base64FromBytes',
         };
         UTILS.bytesFromBase64 = {
             type: 'import',
             path: '@endo/base64',
-            name: 'encodeBase64',
+            name: 'decodeBase64',
             importAs: 'bytesFromBase64',
         };
     }
diff --git a/module/helpers/internal.js b/module/helpers/internal.js
index d24ec7055da9fa30441986a382faab7186ce3cdc..2fb7cc4167b45537ae145bda146f779bc09174bb 100644
--- a/module/helpers/internal.js
+++ b/module/helpers/internal.js
@@ -11,39 +11,8 @@ if (_m0.util.Long !== Long) {
 
 export { Long };
 
-declare var self: any | undefined;
-declare var window: any | undefined;
-declare var global: any | undefined;
-var globalThis: any = (() => {
-    if (typeof globalThis !== 'undefined') return globalThis;
-    if (typeof self !== 'undefined') return self;
-    if (typeof window !== 'undefined') return window;
-    if (typeof global !== 'undefined') return global;
-    throw 'Unable to locate global object';
-})();
-
-const atob: (b64: string) => string =
-    globalThis.atob || ((b64) => globalThis.Buffer.from(b64, 'base64').toString('binary'));
-
-export function bytesFromBase64(b64: string): Uint8Array {
-    const bin = atob(b64);
-    const arr = new Uint8Array(bin.length);
-    for (let i = 0; i < bin.length; ++i) {
-        arr[i] = bin.charCodeAt(i);
-    }
-    return arr;
-}
-
-const btoa: (bin: string) => string =
-    globalThis.btoa || ((bin) => globalThis.Buffer.from(bin, 'binary').toString('base64'));
-
-export function base64FromBytes(arr: Uint8Array): string {
-    const bin: string[] = [];
-    arr.forEach((byte) => {
-        bin.push(String.fromCharCode(byte));
-    });
-    return btoa(bin.join(''));
-}
+// PATCH: Use base64Lib: 'endo' instead for bytesFromBase64 and base64FromBytes
+// TODO update Telescope to put these helpers in a separate module so 'helpers' doesn't entail ambient authority
 
 export interface AminoHeight {
     readonly revision_number?: string;
diff --git a/src/generators/customize-utils.ts b/src/generators/customize-utils.ts
index 5373663e63ee9e99ee995f3858065aab27008f41..bb8728372b5824c03297afe5859760d6e6d1cee4 100644
--- a/src/generators/customize-utils.ts
+++ b/src/generators/customize-utils.ts
@@ -38,13 +38,13 @@ export const plugin = (builder: TelescopeBuilder) => {
     UTILS.base64FromBytes = {
       type: 'import',
       path: '@endo/base64',
-      name: 'decodeBase64',
+      name: 'encodeBase64',
       importAs: 'base64FromBytes',
     };
     UTILS.bytesFromBase64 = {
       type: 'import',
       path: '@endo/base64',
-      name: 'encodeBase64',
+      name: 'decodeBase64',
       importAs: 'bytesFromBase64',
     };
   } else {
diff --git a/src/helpers/internal-for-bigint.ts b/src/helpers/internal-for-bigint.ts
index 9308f4083739a4cfb9826b0e265a3a6338755dfe..425b6171e137f35c92c21ef8b505e6427c5b7959 100644
--- a/src/helpers/internal-for-bigint.ts
+++ b/src/helpers/internal-for-bigint.ts
@@ -2,41 +2,8 @@ import { TelescopeOptions } from "@cosmology/types";
 
 export const getHelperForBigint = (options: TelescopeOptions) => {
   return `
-declare var self: any | undefined;
-declare var window: any | undefined;
-declare var global: any | undefined;
-var globalThis: any = (() => {
-  if (typeof globalThis !== 'undefined') return globalThis;
-  if (typeof self !== 'undefined') return self;
-  if (typeof window !== 'undefined') return window;
-  if (typeof global !== 'undefined') return global;
-  throw 'Unable to locate global object';
-})();
-
-const atob: (b64: string) => string =
-  globalThis.atob ||
-  ((b64) => globalThis.Buffer.from(b64, 'base64').toString('binary'));
-
-export function bytesFromBase64(b64: string): Uint8Array {
-  const bin = atob(b64);
-  const arr = new Uint8Array(bin.length);
-  for (let i = 0; i < bin.length; ++i) {
-    arr[i] = bin.charCodeAt(i);
-  }
-  return arr;
-}
-
-const btoa: (bin: string) => string =
-  globalThis.btoa ||
-  ((bin) => globalThis.Buffer.from(bin, 'binary').toString('base64'));
-
-export function base64FromBytes(arr: Uint8Array): string {
-  const bin: string[] = [];
-  arr.forEach((byte) => {
-    bin.push(String.fromCharCode(byte));
-  });
-  return btoa(bin.join(''));
-}
+// PATCH: Use base64Lib: 'endo' instead for bytesFromBase64 and base64FromBytes
+// TODO update Telescope to put these helpers in a separate module so 'helpers' doesn't entail ambient authority
 
 export interface AminoHeight {
   readonly revision_number?: string;
diff --git a/src/helpers/internal.ts b/src/helpers/internal.ts
index ff08a4f1dbb035054dc8f308ba1adba3c51c2427..df2b8b6fc99d10fcc463b55b045f8cad4c430199 100644
--- a/src/helpers/internal.ts
+++ b/src/helpers/internal.ts
@@ -13,39 +13,8 @@ if (_m0.util.Long !== Long) {
 
 export { Long };
 
-declare var self: any | undefined;
-declare var window: any | undefined;
-declare var global: any | undefined;
-var globalThis: any = (() => {
-    if (typeof globalThis !== 'undefined') return globalThis;
-    if (typeof self !== 'undefined') return self;
-    if (typeof window !== 'undefined') return window;
-    if (typeof global !== 'undefined') return global;
-    throw 'Unable to locate global object';
-})();
-
-const atob: (b64: string) => string =
-    globalThis.atob || ((b64) => globalThis.Buffer.from(b64, 'base64').toString('binary'));
-
-export function bytesFromBase64(b64: string): Uint8Array {
-    const bin = atob(b64);
-    const arr = new Uint8Array(bin.length);
-    for (let i = 0; i < bin.length; ++i) {
-        arr[i] = bin.charCodeAt(i);
-    }
-    return arr;
-}
-
-const btoa: (bin: string) => string =
-    globalThis.btoa || ((bin) => globalThis.Buffer.from(bin, 'binary').toString('base64'));
-
-export function base64FromBytes(arr: Uint8Array): string {
-    const bin: string[] = [];
-    arr.forEach((byte) => {
-        bin.push(String.fromCharCode(byte));
-    });
-    return btoa(bin.join(''));
-}
+// PATCH: Use base64Lib: 'endo' instead for bytesFromBase64 and base64FromBytes
+// TODO update Telescope to put these helpers in a separate module so 'helpers' doesn't entail ambient authority
 
 export interface AminoHeight {
     readonly revision_number?: string;
