diff --git a/src/index.d.ts b/src/index.d.ts
index 1263928..59a9afc 100644
--- a/src/index.d.ts
+++ b/src/index.d.ts
@@ -22,7 +22,7 @@ export function importBundle(bundle: any, options?: {}, powers?: {}): Promise<an
 export function typedImportBundle(bundle: any, options?: {}, powers?: {}): Promise<any>;
 export function bundleTestExports(exports: Record<PropertyKey, unknown>): {
     [x: symbol]: Record<PropertyKey, unknown>;
-    moduleFormat: string;
+    moduleFormat: 'test';
 };
 /**
  * The bundle importer is designed for (serializable) source modules but can
diff --git a/src/index.js b/src/index.js
index a12e0ae..d03a11b 100644
--- a/src/index.js
+++ b/src/index.js
@@ -206,6 +206,7 @@ export const typedImportBundle = importBundle;
  * A utility function for producing test bundles, which are not serializable
  * as JSON or passable.
  * @param {Record<PropertyKey, unknown>} exports
+ * @returns {{ moduleFormat: 'test', [x: symbol]: Record<PropertyKey, unknown> }}
  */
 export const bundleTestExports = exports => {
   const symbols = Object.getOwnPropertySymbols(exports).filter(
