diff --git a/cjs/visitor.js b/cjs/visitor.js
index 98e61e92077976a2c0761102cf4d8c8eb325a63e..4c918f045a71d5df2ef606dfc714076289502833 100644
--- a/cjs/visitor.js
+++ b/cjs/visitor.js
@@ -64,6 +64,7 @@ class GraphQLRequestVisitor extends visitor_plugin_common_1.ClientSideBaseVisito
             .map(o => {
             const operationType = o.node.operation;
             const operationName = o.node.name.value;
+            const operationDocComment = (0, visitor_plugin_common_1.transformComment)(o.node.description);
             const optionalVariables = !o.node.variableDefinitions ||
                 o.node.variableDefinitions.length === 0 ||
                 o.node.variableDefinitions.every(v => v.type.kind !== graphql_1.Kind.NON_NULL_TYPE || v.defaultValue);
@@ -74,11 +75,11 @@ class GraphQLRequestVisitor extends visitor_plugin_common_1.ClientSideBaseVisito
                     docArg = `${docVarName}String`;
                     extraVariables.push(`const ${docArg} = print(${docVarName});`);
                 }
-                return `${operationName}(variables${optionalVariables ? '?' : ''}: ${o.operationVariablesTypes}, requestHeaders?: GraphQLClientRequestHeaders): Promise<{ data: ${o.operationResultType}; errors?: GraphQLError[]; extensions?: ${this.config.extensionsType}; headers: Headers; status: number; }> {
+                return `${operationDocComment}${operationName}(variables${optionalVariables ? '?' : ''}: ${o.operationVariablesTypes}, requestHeaders?: GraphQLClientRequestHeaders): Promise<{ data: ${o.operationResultType}; errors?: GraphQLError[]; extensions?: ${this.config.extensionsType}; headers: Headers; status: number; }> {
     return withWrapper((wrappedRequestHeaders) => client.rawRequest<${o.operationResultType}>(${docArg}, variables, {...requestHeaders, ...wrappedRequestHeaders}), '${operationName}', '${operationType}', variables);
 }`;
             }
-            return `${operationName}(variables${optionalVariables ? '?' : ''}: ${o.operationVariablesTypes}, requestHeaders?: GraphQLClientRequestHeaders, signal?: RequestInit['signal']): Promise<${o.operationResultType}> {
+            return `${operationDocComment}${operationName}(variables${optionalVariables ? '?' : ''}: ${o.operationVariablesTypes}, requestHeaders?: GraphQLClientRequestHeaders, signal?: RequestInit['signal']): Promise<${o.operationResultType}> {
   return withWrapper((wrappedRequestHeaders) => client.request<${o.operationResultType}>({ document: ${docVarName}, variables, requestHeaders: { ...requestHeaders, ...wrappedRequestHeaders }, signal }), '${operationName}', '${operationType}', variables);
 }`;
         })
diff --git a/esm/visitor.js b/esm/visitor.js
index f2481ee2e1f6082314b4045b161b5882568902c7..56bec30c9b41ec5747945c991c59c3d724468877 100644
--- a/esm/visitor.js
+++ b/esm/visitor.js
@@ -1,6 +1,6 @@
 import autoBind from 'auto-bind';
 import { Kind, print } from 'graphql';
-import { ClientSideBaseVisitor, DocumentMode, getConfigValue, indentMultiline, } from '@graphql-codegen/visitor-plugin-common';
+import { ClientSideBaseVisitor, DocumentMode, getConfigValue, indentMultiline, transformComment, } from '@graphql-codegen/visitor-plugin-common';
 const additionalExportedTypes = `
 export type SdkFunctionWrapper = <T>(action: (requestHeaders?:Record<string, string>) => Promise<T>, operationName: string, operationType?: string, variables?: any) => Promise<T>;
 `;
@@ -60,6 +60,7 @@ export class GraphQLRequestVisitor extends ClientSideBaseVisitor {
             .map(o => {
             const operationType = o.node.operation;
             const operationName = o.node.name.value;
+            const operationDocComment = transformComment(o.node.description);
             const optionalVariables = !o.node.variableDefinitions ||
                 o.node.variableDefinitions.length === 0 ||
                 o.node.variableDefinitions.every(v => v.type.kind !== Kind.NON_NULL_TYPE || v.defaultValue);
@@ -70,11 +71,11 @@ export class GraphQLRequestVisitor extends ClientSideBaseVisitor {
                     docArg = `${docVarName}String`;
                     extraVariables.push(`const ${docArg} = print(${docVarName});`);
                 }
-                return `${operationName}(variables${optionalVariables ? '?' : ''}: ${o.operationVariablesTypes}, requestHeaders?: GraphQLClientRequestHeaders): Promise<{ data: ${o.operationResultType}; errors?: GraphQLError[]; extensions?: ${this.config.extensionsType}; headers: Headers; status: number; }> {
+                return `${operationDocComment}${operationName}(variables${optionalVariables ? '?' : ''}: ${o.operationVariablesTypes}, requestHeaders?: GraphQLClientRequestHeaders): Promise<{ data: ${o.operationResultType}; errors?: GraphQLError[]; extensions?: ${this.config.extensionsType}; headers: Headers; status: number; }> {
     return withWrapper((wrappedRequestHeaders) => client.rawRequest<${o.operationResultType}>(${docArg}, variables, {...requestHeaders, ...wrappedRequestHeaders}), '${operationName}', '${operationType}', variables);
 }`;
             }
-            return `${operationName}(variables${optionalVariables ? '?' : ''}: ${o.operationVariablesTypes}, requestHeaders?: GraphQLClientRequestHeaders, signal?: RequestInit['signal']): Promise<${o.operationResultType}> {
+            return `${operationDocComment}${operationName}(variables${optionalVariables ? '?' : ''}: ${o.operationVariablesTypes}, requestHeaders?: GraphQLClientRequestHeaders, signal?: RequestInit['signal']): Promise<${o.operationResultType}> {
   return withWrapper((wrappedRequestHeaders) => client.request<${o.operationResultType}>({ document: ${docVarName}, variables, requestHeaders: { ...requestHeaders, ...wrappedRequestHeaders }, signal }), '${operationName}', '${operationType}', variables);
 }`;
         })
