diff --git a/dist/conventional-commits/recommend-version.js b/dist/conventional-commits/recommend-version.js
index 7c16eab1865e2913a3b3cd1407cc866a9ed448af..d26791f12884978cb426e2ed798a93b2006f0be9 100644
--- a/dist/conventional-commits/recommend-version.js
+++ b/dist/conventional-commits/recommend-version.js
@@ -46,6 +46,14 @@ export async function recommendVersion(pkg, type, recommendationOptions, premajo
             const whatBumpFn = (await getWhatBump());
             const bumpResult = (await bumper.bump(whatBumpFn));
             let releaseType = (bumpResult?.releaseType || 'patch');
+            if (semver.major(pkg.version) === 0) {
+                if (releaseType === 'major') {
+                    releaseType = 'minor';
+                }
+                else if (premajorVersionBump === 'force-patch') {
+                    releaseType = 'patch';
+                }
+            }
             if (prereleaseId) {
                 const shouldBump = conventionalBumpPrerelease || shouldBumpPrerelease(releaseType, pkg.version);
                 const prereleaseType = (shouldBump ? `pre${releaseType}` : 'prerelease');
@@ -53,14 +61,6 @@ export async function recommendVersion(pkg, type, recommendationOptions, premajo
                 resolve(applyBuildMetadata(semver.inc(pkg.version, prereleaseType, prereleaseId), buildMetadata));
             }
             else {
-                if (semver.major(pkg.version) === 0) {
-                    if (releaseType === 'major') {
-                        releaseType = 'minor';
-                    }
-                    else if (premajorVersionBump === 'force-patch') {
-                        releaseType = 'patch';
-                    }
-                }
                 log.verbose(type, 'increment %s by %s - %s', pkg.version, releaseType, pkg.name);
                 resolve(applyBuildMetadata(semver.inc(pkg.version, releaseType), buildMetadata));
             }
