diff --git a/index.d.ts b/index.d.ts
index fee3cf32613356eafc78749ce682881823d77331..2a6eab29ff5eff599809a4470eb7e6ee0ebdb6b8 100755
--- a/index.d.ts
+++ b/index.d.ts
@@ -26,7 +26,7 @@ declare namespace G {
     let Glob: IGlobStatic;
     let GlobSync: IGlobSyncStatic;
 
-    interface IOptions extends minimatch.IOptions {
+    interface IOptions extends minimatch.MinimatchOptions {
         cwd?: string | undefined;
         root?: string | undefined;
         dot?: boolean | undefined;
@@ -72,8 +72,8 @@ declare namespace G {
     }
 
     interface IGlobBase {
-        minimatch: minimatch.IMinimatch;
-        options: IOptions;
+        minimatch: minimatch.Minimatch;
+        options: minimatch.MinimatchOptions;
         aborted: boolean;
         cache: { [path: string]: boolean | 'DIR' | 'FILE' | ReadonlyArray<string> };
         statCache: { [path: string]: false | { isDirectory(): boolean; } | undefined };
