diff --git a/index.js b/index.js
new file mode 100644
index 0000000000000000000000000000000000000000..bee62d84afa43a1a68e1b6ac74ebdea6831fefcb
--- /dev/null
+++ b/index.js
@@ -0,0 +1,2 @@
+// XXX work around https://github.com/import-js/eslint-plugin-import/issues/1810
+export {default} from './lib/worker/main.cjs';
diff --git a/lib/cli.js b/lib/cli.js
index b613666c8946f7bdd61e3ef84347e13cf50fff95..fbed44368e394454bf146a8b5bd35ad87f4d9087 100644
--- a/lib/cli.js
+++ b/lib/cli.js
@@ -448,6 +448,10 @@ export default async function loadCli() { // eslint-disable-line complexity
 		workerArgv: argv['--'],
 	});
 
+	if (process.env.AGORIC_AVA_USE_TAP){
+		combined.tap = true;
+	}
+
 	const reporter = combined.tap && !combined.watch && debug === null ? new TapReporter({
 		extensions: globs.extensions,
 		projectDir,
diff --git a/lib/fork.js b/lib/fork.js
index 7630baa396ee8c515a82ea98ece67e4d8daa5491..78ced7746c904fc7976ab2f17da65fc9eaf036d1 100644
--- a/lib/fork.js
+++ b/lib/fork.js
@@ -7,6 +7,7 @@ import Emittery from 'emittery';
 import {pEvent} from 'p-event';
 
 import {controlFlow} from './ipc-flow-control.cjs';
+import {setCappedTimeout} from './now-and-timers.cjs';
 import serializeError from './serialize-error.js';
 
 let workerPath = new URL('worker/base.js', import.meta.url);
@@ -91,6 +92,11 @@ export default function loadFork(file, options, execArgv = process.execArgv) {
 	});
 
 	let forcedExit = false;
+	let exitCode = null;
+	const exit = () => {
+		forcedExit = true;
+		close();
+	};
 	const send = evt => {
 		if (!finished && !forcedExit) {
 			postMessage({ava: evt});
@@ -132,6 +138,12 @@ export default function loadFork(file, options, execArgv = process.execArgv) {
 					break;
 				}
 
+				case 'exiting': {
+					exitCode = message.ava.code;
+					setCappedTimeout(() => finished || exit(), 10_000).unref();
+					break;
+				}
+
 				default: {
 					emitStateChange(message.ava);
 				}
@@ -145,9 +157,20 @@ export default function loadFork(file, options, execArgv = process.execArgv) {
 
 		worker.on('exit', (code, signal) => {
 			if (forcedExit) {
-				emitStateChange({type: 'worker-finished', forcedExit});
-			} else if (code > 0) {
-				emitStateChange({type: 'worker-failed', nonZeroExitCode: code});
+				if (exitCode === null) {
+					emitStateChange({type: 'worker-finished', forcedExit});
+				} else if (!exitCode) {
+					if (!process.env.NODE_V8_COVERAGE) {
+						emitStateChange({type: 'worker-failed', err: Error('Test did not cleanup'), signal: 'exit timeout'});
+					} else {
+						emitStateChange({type: 'worker-stderr', chunk: `Test did not cleanup, ignoring because NODE_V8_COVERAGE is set.\n`});
+						emitStateChange({type: 'worker-finished', forcedExit: false});
+					}
+				} else {
+					emitStateChange({type: 'worker-failed', nonZeroExitCode: exitCode});
+				}
+			} else if (code > 0 || exitCode > 0) {
+				emitStateChange({type: 'worker-failed', nonZeroExitCode: code || exitCode});
 			} else if (code === null && signal) {
 				emitStateChange({type: 'worker-failed', signal});
 			} else {
@@ -163,10 +186,7 @@ export default function loadFork(file, options, execArgv = process.execArgv) {
 		threadId: worker.threadId,
 		promise,
 
-		exit() {
-			forcedExit = true;
-			close();
-		},
+		exit,
 
 		notifyOfPeerFailure() {
 			send({type: 'peer-failed'});
diff --git a/lib/reporters/default.js b/lib/reporters/default.js
index 804e285cc3322fab0e8fa17851800546d42b499f..8daf5ee5b141c70fcebadb4d0a4f06be1cb75120 100644
--- a/lib/reporters/default.js
+++ b/lib/reporters/default.js
@@ -678,8 +678,11 @@ export default class Reporter {
 			this.lineWriter.writeLine(colors.todo(`${this.stats.todoTests} ${plur('test', this.stats.todoTests)} todo`));
 		}
 
-		if (this.stats.unhandledRejections > 0) {
-			this.lineWriter.writeLine(colors.error(`${this.stats.unhandledRejections} unhandled ${plur('rejection', this.stats.unhandledRejections)}`));
+		// AGORIC: TODO(#11028): unhandled exceptions patch should be accomplished in userspace.
+    if (this.stats.unhandledRejections > 0 || process.env.AGORIC_AVA_EXPECT_UNHANDLED_REJECTIONS) {
+			const expected = Number(process.env.AGORIC_AVA_EXPECT_UNHANDLED_REJECTIONS || 0)
+			const color = this.stats.unhandledRejections === expected ? 'skip' : 'error';
+			this.lineWriter.writeLine(colors[color](`${this.stats.unhandledRejections} unhandled ${plur('rejection', this.stats.unhandledRejections)} (AGORIC_AVA_EXPECT_UNHANDLED_REJECTIONS=${expected})`));
 		}
 
 		if (this.stats.uncaughtExceptions > 0) {
diff --git a/lib/reporters/tap.js b/lib/reporters/tap.js
index b1989a4d60b4f224c01e9fb285e9f789848593f0..fa1617f7000f760e781741c31d30101a5361baa0 100644
--- a/lib/reporters/tap.js
+++ b/lib/reporters/tap.js
@@ -45,6 +45,10 @@ function dumpError(error) {
 	return object;
 }
 
+function durationOut(dur) {
+	return dur ? ` %ava-dur=${dur}ms` : '';
+}
+
 export default class TapReporter {
 	constructor(options) {
 		this.i = 0;
@@ -96,6 +100,7 @@ export default class TapReporter {
 	}
 
 	writeTest(evt, flags) {
+ 		evt.title += durationOut(evt.duration);
 		this.reportStream.write(supertap.test(this.prefixTitle(evt.testFile, evt.title), {
 			comment: evt.logs,
 			error: evt.err ? dumpError(evt.err) : null,
@@ -108,7 +113,7 @@ export default class TapReporter {
 
 	writeCrash(evt, title) {
 		this.crashCount++;
-		this.reportStream.write(supertap.test(title || evt.err.summary || evt.type, {
+		this.reportStream.write(supertap.test((title || evt.err.summary || evt.type) + durationOut(evt.duration), {
 			comment: evt.logs,
 			error: evt.err ? dumpError(evt.err) : null,
 			index: ++this.i,
diff --git a/lib/run-status.js b/lib/run-status.js
index 84713456a0b9c4c77ca8a6cced38406d43eb4829..74337125494f638841249351c62f59202b85f681 100644
--- a/lib/run-status.js
+++ b/lib/run-status.js
@@ -228,7 +228,7 @@ export default class RunStatus extends Emittery {
 			|| this.stats.sharedWorkerErrors > 0
 			|| this.stats.timeouts > 0
 			|| this.stats.uncaughtExceptions > 0
-			|| this.stats.unhandledRejections > 0
+      // AGORIC: `unhandledRejections` are processed below.
 		) {
 			return 1;
 		}
@@ -237,6 +237,14 @@ export default class RunStatus extends Emittery {
 			return 1;
 		}
 
+		// AGORIC: TODO(#11028): unhandled exceptions patch should be accomplished in userspace.
+		if (this.stats.unhandledRejections > 0 || process.env.AGORIC_AVA_EXPECT_UNHANDLED_REJECTIONS) {
+			const expected = Number(process.env.AGORIC_AVA_EXPECT_UNHANDLED_REJECTIONS || 0);
+			if (this.stats.unhandledRejections !== expected) {
+				return 1;
+			}
+		}
+
 		return 0;
 	}
 
diff --git a/lib/worker/base.js b/lib/worker/base.js
index cdd3c4a1a254964102f042244f3aa0a8b4b9a68f..4302a4ad0ed66571f15f7df73c549155a8d294ee 100644
--- a/lib/worker/base.js
+++ b/lib/worker/base.js
@@ -1,5 +1,6 @@
 import {createRequire} from 'node:module';
 import process from 'node:process';
+import v8 from 'node:v8';
 import {pathToFileURL} from 'node:url';
 import {workerData} from 'node:worker_threads';
 
@@ -28,12 +29,17 @@ const realExit = process.exit;
 
 async function exit(code, forceSync = false) {
 	dependencyTracking.flush();
+	if (process.env.NODE_V8_COVERAGE) try {
+		v8.takeCoverage();
+	} catch(err) {}
+	channel.send({type: 'exiting', code});
 	const flushing = channel.flush();
 	if (!forceSync) {
 		await flushing;
+		process.exitCode ||= code;
+	} else {
+		apply(realExit, process, [code]);
 	}
-
-	apply(realExit, process, [code]);
 }
 
 const handleProcessExit = (fn, receiver, args) => {
