diff --git a/node_modules/@confio/ics23/node_modules/protobufjs/src/util/minimal.js b/node_modules/@confio/ics23/node_modules/protobufjs/src/util/minimal.js
index 3c406de..083f71c 100644
--- a/node_modules/@confio/ics23/node_modules/protobufjs/src/util/minimal.js
+++ b/node_modules/@confio/ics23/node_modules/protobufjs/src/util/minimal.js
@@ -280,13 +280,38 @@ function newError(name) {
             merge(this, properties);
     }
 
-    (CustomError.prototype = Object.create(Error.prototype)).constructor = CustomError;
-
-    Object.defineProperty(CustomError.prototype, "name", { get: function() { return name; } });
-
-    CustomError.prototype.toString = function toString() {
-        return this.name + ": " + this.message;
-    };
+    CustomError.prototype = Object.create(Error.prototype, {
+        constructor: {
+            value: CustomError,
+            writable: true,
+            // enumerable: true would accurately preserve the behavior of the
+            // original assignment, but I'm guessing that was not intentional.
+            // For an actual error subclass, this property would not
+            // be enumerable.
+            enumerable: false,
+            configurable: true,
+        },
+        name: {
+            get() { return name; },
+            set: undefined,
+            enumerable: false,
+            // configurable: false would accurately preserve the behavior of
+            // the original, but I'm guessing that was not intentional.
+            // For an actual error subclass, this property would
+            // be configurable.
+            configurable: true,
+        },
+        toString: {
+            value() { return this.name + ": " + this.message; },
+            writable: true,
+            // enumerable: true would accurately preserve the behavior of the
+            // original assignment, but I'm guessing that was not intentional.
+            // For an actual error subclass, this property would not
+            // be enumerable.
+            enumerable: false,
+            configurable: true,
+        },
+    });
 
     return CustomError;
 }
