diff --git a/node_modules/@endo/bundle-source/src/tool.js b/node_modules/@endo/bundle-source/src/tool.js
index fd9088e..43233d2 100755
--- a/node_modules/@endo/bundle-source/src/tool.js
+++ b/node_modules/@endo/bundle-source/src/tool.js
@@ -37,7 +37,7 @@ export const makeFileWriter = (fileName, { fs, path }) => {
 export const makeBundleCache = (wr, cwd, readPowers, opts) => {
   const {
     toBundleName = n => `bundle-${n}.js`,
-    toBundleMeta = n => `bundle-${n}-meta.js`,
+    toBundleMeta = n => `bundle-${n}-meta.json`,
   } = opts || {};
 
   const add = async (rootPath, targetName) => {
