diff --git a/node_modules/@endo/exo/index.d.ts b/node_modules/@endo/exo/index.d.ts
index bd4c7ed..463ae5e 100644
--- a/node_modules/@endo/exo/index.d.ts
+++ b/node_modules/@endo/exo/index.d.ts
@@ -1,3 +1,3 @@
 export { GET_INTERFACE_GUARD } from "./src/get-interface.js";
-export { initEmpty, defineExoClass, defineExoClassKit, makeExo } from "./src/exo-makers.js";
+export * from "./src/exo-makers.js";
 //# sourceMappingURL=index.d.ts.map
diff --git a/node_modules/@endo/exo/index.js b/node_modules/@endo/exo/index.js
index 0936e29..3924dd4 100644
--- a/node_modules/@endo/exo/index.js
+++ b/node_modules/@endo/exo/index.js
@@ -1,8 +1,3 @@
-export {
-  initEmpty,
-  defineExoClass,
-  defineExoClassKit,
-  makeExo,
-} from './src/exo-makers.js';
+export * from './src/exo-makers.js';
 
 export { GET_INTERFACE_GUARD } from './src/get-interface.js';
diff --git a/node_modules/@endo/exo/tools.d.ts b/node_modules/@endo/exo/tools.d.ts
index 0537b0b..be5d131 100644
--- a/node_modules/@endo/exo/tools.d.ts
+++ b/node_modules/@endo/exo/tools.d.ts
@@ -1,2 +1,2 @@
-export { defendPrototype, defendPrototypeKit } from "./src/exo-tools.js";
+export * from "./src/exo-tools.js";
 //# sourceMappingURL=tools.d.ts.map
diff --git a/node_modules/@endo/exo/tools.js b/node_modules/@endo/exo/tools.js
index e8db521..072ff15 100644
--- a/node_modules/@endo/exo/tools.js
+++ b/node_modules/@endo/exo/tools.js
@@ -1 +1 @@
-export { defendPrototype, defendPrototypeKit } from './src/exo-tools.js';
+export * from './src/exo-tools.js';
