diff --git a/node_modules/@endo/captp/src/finalize.js b/node_modules/@endo/captp/src/finalize.js
index 871bcf5..60b5edf 100644
--- a/node_modules/@endo/captp/src/finalize.js
+++ b/node_modules/@endo/captp/src/finalize.js
@@ -99,7 +99,14 @@ export const makeFinalizingMap = (finalizer, opts) => {
     },
     // Does deref, and thus does guarantee stability of the value until the
     // end of the turn.
-    get: key => keyToRef.get(key)?.deref(),
+    // Prefer: get: key => keyToRef.get(key)?.deref(),
+    get: key => {
+      const wr = keyToRef.get(key);
+      if (!wr) {
+        return wr;
+      }
+      return wr.deref();
+    },
     has: key => finalizingMap.get(key) !== undefined,
     // Does deref, and thus does guarantee stability of both old and new values
     // until the end of the turn.
