diff --git a/node_modules/@lerna/conventional-commits/lib/recommend-version.js b/node_modules/@lerna/conventional-commits/lib/recommend-version.js
index a982adf..453ef14 100644
--- a/node_modules/@lerna/conventional-commits/lib/recommend-version.js
+++ b/node_modules/@lerna/conventional-commits/lib/recommend-version.js
@@ -58,6 +58,15 @@ function recommendVersion(pkg, type, { changelogPreset, rootPath, tagPrefix, pre
         // we still need to bump _something_ because lerna saw a change here
         let releaseType = data.releaseType || "patch";
 
+        // Don't gratuitously break compatibility with clients using `^0.x.y`.
+        if (semver.major(pkg.version) === 0) {
+          if (releaseType === "major") {
+            releaseType = "minor";
+          } else if (releaseType === "minor") {
+            releaseType = "patch";
+          }
+        }
+
         if (prereleaseId) {
           const shouldBump = shouldBumpPrerelease(releaseType, pkg.version);
           const prereleaseType = shouldBump ? `pre${releaseType}` : "prerelease";
