diff --git a/node_modules/cosmjs-types/node_modules/protobufjs/src/util/minimal.js b/node_modules/cosmjs-types/node_modules/protobufjs/src/util/minimal.js
index 3c406de..fb88de4 100644
--- a/node_modules/cosmjs-types/node_modules/protobufjs/src/util/minimal.js
+++ b/node_modules/cosmjs-types/node_modules/protobufjs/src/util/minimal.js
@@ -280,13 +280,26 @@ function newError(name) {
             merge(this, properties);
     }
 
-    (CustomError.prototype = Object.create(Error.prototype)).constructor = CustomError;
-
-    Object.defineProperty(CustomError.prototype, "name", { get: function() { return name; } });
-
-    CustomError.prototype.toString = function toString() {
-        return this.name + ": " + this.message;
-    };
+    CustomError.prototype = Object.create(Error.prototype, {
+        constructor: {
+            value: CustomError,
+            writable: true,
+            enumerable: false,
+            configurable: true,
+        },
+        name: {
+            get() { return name; },
+            set: undefined,
+            enumerable: false,
+            configurable: true,
+        },
+        toString: {
+            value() { return this.name + ": " + this.message; },
+            writable: true,
+            enumerable: false,
+            configurable: true,
+        },
+    });
 
     return CustomError;
 }
