diff --git a/node_modules/express/lib/router/index.js b/node_modules/express/lib/router/index.js
index abb3a6f..be10092 100644
--- a/node_modules/express/lib/router/index.js
+++ b/node_modules/express/lib/router/index.js
@@ -515,12 +515,17 @@ proto.route = function route(path) {
 };
 
 // create Router#VERB functions
-methods.concat('all').forEach(function(method){
-  proto[method] = function(path){
-    var route = this.route(path)
-    route[method].apply(route, slice.call(arguments, 1));
-    return this;
-  };
+methods.concat('all').forEach(function (method) {
+  Object.defineProperty(proto, method, {
+    value: function (path) {
+      var route = this.route(path)
+      route[method].apply(route, slice.call(arguments, 1));
+      return this;
+    },
+    writable: true,
+    enumerable: true,
+    configurable: true
+  });
 });
 
 // append methods to a list of methods
