diff --git a/node_modules/readable-stream/errors.js b/node_modules/readable-stream/errors.js
index 8471526..a4169da 100644
--- a/node_modules/readable-stream/errors.js
+++ b/node_modules/readable-stream/errors.js
@@ -21,7 +21,18 @@ function createErrorType(code, message, Base) {
     }
   }
 
-  NodeError.prototype.name = Base.name;
+  Object.defineProperties(NodeError.prototype, {
+    name: {
+      value: Base.name,
+      writable: true,
+      // enumerable: true would accurately preserve the behavior of the
+      // original assignment, but I'm guessing that was not intentional.
+      // For an actual error subclass, this property would not
+      // be enumerable.
+      enumerable: false,
+      configurable: true,
+    }
+  })
   NodeError.prototype.code = code;
 
   codes[code] = NodeError;
