diff --git a/node_modules/ses/dist/ses.cjs b/node_modules/ses/dist/ses.cjs
index 121da57..107c4ca 100644
--- a/node_modules/ses/dist/ses.cjs
+++ b/node_modules/ses/dist/ses.cjs
@@ -9808,7 +9808,7 @@ markVirtualizedNativeFunction)
           // of a debugging attempt. See the comment at top of file for an
           // explanation.
           // eslint-disable-next-line @endo/no-polymorphic-call
-          console.log(`Removing ${subPath}`);
+          console.warn(`Removing ${subPath}`);
         }
         try {
           delete obj[prop];
@@ -10051,7 +10051,7 @@ options = {}) =>
 
   if (seemsToBeLockedDown()) {
     // eslint-disable-next-line @endo/no-polymorphic-call
-    console.log('Seems to already be locked down. Skipping second lockdown');
+    console.warn('Seems to already be locked down. Skipping second lockdown');
     return alreadyHardenedIntrinsics;
   }
 
diff --git a/node_modules/ses/dist/ses.umd.js b/node_modules/ses/dist/ses.umd.js
index 121da57..107c4ca 100644
--- a/node_modules/ses/dist/ses.umd.js
+++ b/node_modules/ses/dist/ses.umd.js
@@ -9808,7 +9808,7 @@ markVirtualizedNativeFunction)
           // of a debugging attempt. See the comment at top of file for an
           // explanation.
           // eslint-disable-next-line @endo/no-polymorphic-call
-          console.log(`Removing ${subPath}`);
+          console.warn(`Removing ${subPath}`);
         }
         try {
           delete obj[prop];
@@ -10051,7 +10051,7 @@ options = {}) =>
 
   if (seemsToBeLockedDown()) {
     // eslint-disable-next-line @endo/no-polymorphic-call
-    console.log('Seems to already be locked down. Skipping second lockdown');
+    console.warn('Seems to already be locked down. Skipping second lockdown');
     return alreadyHardenedIntrinsics;
   }
 
