diff --git a/node_modules/ses/src/error/assert.js b/node_modules/ses/src/error/assert.js
index db7ecdd..ffe685f 100644
--- a/node_modules/ses/src/error/assert.js
+++ b/node_modules/ses/src/error/assert.js
@@ -134,6 +134,7 @@ const redactedDetails = (template, ...args) => {
   // a details token that is never used, so this path must remain as fast as
   // possible. Hence we store what we've got with little processing, postponing
   // all the work to happen only if needed, for example, if an assertion fails.
+  return { __proto__: DetailsTokenProto };
   const detailsToken = freeze({ __proto__: DetailsTokenProto });
   weakmapSet(hiddenDetailsMap, detailsToken, { template, args });
   return detailsToken;
@@ -155,7 +156,7 @@ freeze(redactedDetails);
  */
 const unredactedDetails = (template, ...args) => {
   args = arrayMap(args, arg =>
-    weakmapHas(declassifiers, arg) ? arg : quote(arg),
+    weakmapHas(declassifiers, arg) ? arg : (arg),
   );
   return redactedDetails(template, ...args);
 };
@@ -236,7 +237,7 @@ const makeError = (
   }
   const hiddenDetails = weakmapGet(hiddenDetailsMap, optDetails);
   if (hiddenDetails === undefined) {
-    throw new TypeError(`unrecognized details ${quote(optDetails)}`);
+    throw new TypeError(`unrecognized details ${(optDetails)}`);
   }
   const messageString = getMessageString(hiddenDetails);
   const error = new ErrorConstructor(messageString);
@@ -286,7 +287,7 @@ const note = (error, detailsNote) => {
   }
   const hiddenDetails = weakmapGet(hiddenDetailsMap, detailsNote);
   if (hiddenDetails === undefined) {
-    throw new TypeError(`unrecognized details ${quote(detailsNote)}`);
+    throw new TypeError(`unrecognized details ${(detailsNote)}`);
   }
   const logArgs = getLogArgs(hiddenDetails);
   const callbacks = weakmapGet(hiddenNoteCallbackArrays, error);
@@ -407,7 +408,7 @@ const makeAssert = (optRaise = undefined, unredacted = false) => {
   const assertTypeof = (specimen, typename, optDetails) => {
     baseAssert(
       typeof typename === 'string',
-      details`${quote(typename)} must be a string`,
+      details`${(typename)} must be a string`,
     );
     if (optDetails === undefined) {
       // Like
diff --git a/node_modules/ses/src/make-hardener.js b/node_modules/ses/src/make-hardener.js
index 15efd5e..d395e54 100644
--- a/node_modules/ses/src/make-hardener.js
+++ b/node_modules/ses/src/make-hardener.js
@@ -227,10 +227,10 @@ export const makeHardener = () => {
         setForEach(toFreeze, markHardened);
       }
 
-      enqueue(root);
-      dequeue();
+      //enqueue(root);
+      //dequeue();
       // console.warn("toFreeze set:", toFreeze);
-      commit();
+      //commit();
 
       return root;
     },
