diff --git a/node_modules/vega-util/build/vega-util.js b/node_modules/vega-util/build/vega-util.js
index 0b40c91..bd0a2e4 100644
--- a/node_modules/vega-util/build/vega-util.js
+++ b/node_modules/vega-util/build/vega-util.js
@@ -611,7 +611,12 @@
 
   function inherits (child, parent, members) {
     const proto = child.prototype = Object.create(parent.prototype);
-    proto.constructor = child;
+    Object.defineProperty(proto, 'constructor', {
+      value: child,
+      writable: true,
+      enumerable: true,
+      configurable: true
+    });
     return extend(proto, members);
   }
 
@@ -859,7 +864,6 @@
   exports.fastmap = fastmap;
   exports.field = field;
   exports.flush = flush;
-  exports.hasOwnProperty = has;
   exports.id = id;
   exports.identity = identity;
   exports.inherits = inherits;
@@ -895,7 +899,6 @@
   exports.toDate = toDate;
   exports.toNumber = toNumber;
   exports.toSet = toSet;
-  exports.toString = toString;
   exports.truncate = truncate;
   exports.truthy = truthy;
   exports.utcquarter = utcquarter;
@@ -907,6 +910,21 @@
   exports.zoomPow = zoomPow;
   exports.zoomSymlog = zoomSymlog;
 
+  Object.defineProperties(exports, {
+    hasOwnProperty: {
+      value: has,
+      writable: true,
+      enumerable: true,
+      configurable: true
+    },
+    toString: {
+      value: toString,
+      writable: true,
+      enumerable: true,
+      configurable: true
+    }
+  });
+
   Object.defineProperty(exports, '__esModule', { value: true });
 
 })));
