diff --git a/dist/node/axios.cjs b/dist/node/axios.cjs
index 9099d870682546f7b5a213a19459723817ec571a..311d5da7afe4379caa85637ac3dd45bbd301035a 100644
--- a/dist/node/axios.cjs
+++ b/dist/node/axios.cjs
@@ -370,9 +370,9 @@ function merge(/* obj1, obj2, obj3, ... */) {
 const extend = (a, b, thisArg, {allOwnKeys}= {}) => {
   forEach(b, (val, key) => {
     if (thisArg && isFunction(val)) {
-      a[key] = bind(val, thisArg);
+      Object.defineProperty(a, key, {value: bind(val, thisArg)});
     } else {
-      a[key] = val;
+      Object.defineProperty(a, key, {value: val});
     }
   }, {allOwnKeys});
   return a;
@@ -403,7 +403,9 @@ const stripBOM = (content) => {
  */
 const inherits = (constructor, superConstructor, props, descriptors) => {
   constructor.prototype = Object.create(superConstructor.prototype, descriptors);
-  constructor.prototype.constructor = constructor;
+  Object.defineProperty(constructor, 'constructor', {
+    value: constructor
+  });  
   Object.defineProperty(constructor, 'super', {
     value: superConstructor.prototype
   });
@@ -565,12 +567,14 @@ const isRegExp = kindOfTest('RegExp');
 
 const reduceDescriptors = (obj, reducer) => {
   const descriptors = Object.getOwnPropertyDescriptors(obj);
-  const reducedDescriptors = {};
+  let reducedDescriptors = {};
 
   forEach(descriptors, (descriptor, name) => {
     let ret;
     if ((ret = reducer(descriptor, name, obj)) !== false) {
-      reducedDescriptors[name] = ret || descriptor;
+      reducedDescriptors = {...reducedDescriptors, 
+        [name]: ret || descriptor
+      };  
     }
   });
 
diff --git a/package.json b/package.json
index cf185bda8119e4c3c5f6ce4c86d342d19ebfcd80..395481d06bd9dd6337964adb5c8dd4acbe167e57 100644
--- a/package.json
+++ b/package.json
@@ -214,5 +214,6 @@
     "extends": [
       "@commitlint/config-conventional"
     ]
-  }
+  },
+  "packageManager": "yarn@1.22.22+sha512.a6b2f7906b721bba3d67d4aff083df04dad64c399707841b7acf00f6b133b7ac24255f2652fa22ae3534329dc6180534e98d17432037ff6fd140556e2bb3137e"
 }
