diff --git a/lib/exception.js b/lib/exception.js
index 7f62daaef8bcdb8248da0fd65e7b2880a5c1cf92..4e85f6ac4164b932e738e5d21fdfe9497ccf93f7 100644
--- a/lib/exception.js
+++ b/lib/exception.js
@@ -26,7 +26,6 @@ function YAMLException(reason, mark) {
   // Super constructor
   Error.call(this);
 
-  this.name = 'YAMLException';
   this.reason = reason;
   this.mark = mark;
   this.message = formatError(this, false);
@@ -37,19 +36,34 @@ function YAMLException(reason, mark) {
     Error.captureStackTrace(this, this.constructor);
   } else {
     // FF, IE 10+ and Safari 6+. Fallback for others
-    this.stack = (new Error()).stack || '';
+    this.stack = new Error().stack || "";
   }
 }
 
-
-// Inherit from Error
-YAMLException.prototype = Object.create(Error.prototype);
-YAMLException.prototype.constructor = YAMLException;
-
-
-YAMLException.prototype.toString = function toString(compact) {
-  return this.name + ': ' + formatError(this, compact);
-};
-
+function makeErrorish(errorConstructor, name) {
+  errorConstructor.prototype = Object.create(Error.prototype, {
+    constructor: {
+      value: errorConstructor,
+      writable: true,
+      enumerable: true,
+      configurable: true,
+    },
+    name: {
+      value: name,
+      writable: true,
+      enumerable: true,
+      configurable: true,
+    },
+    toString: {
+      value: function toString(compact) {
+        return this.name + ": " + formatError(this, compact);
+      },
+      writable: true,
+      enumerable: true,
+      configurable: true,
+    },
+  });
+}
+makeErrorish(YAMLException, "YAMLException");
 
 module.exports = YAMLException;
diff --git a/dist/js-yaml.mjs b/dist/js-yaml.mjs
index be71cad18da90d26e66ce43d419880ad008d27e0..b1b98036703a401c170fc0bf2165560748815a83 100644
--- a/dist/js-yaml.mjs
+++ b/dist/js-yaml.mjs
@@ -109,14 +109,32 @@ function YAMLException$1(reason, mark) {
 
 
 // Inherit from Error
-YAMLException$1.prototype = Object.create(Error.prototype);
-YAMLException$1.prototype.constructor = YAMLException$1;
-
-
-YAMLException$1.prototype.toString = function toString(compact) {
-  return this.name + ': ' + formatError(this, compact);
-};
-
+function makeErrorish(errorConstructor, name) {
+  errorConstructor.prototype = Object.create(Error.prototype, {
+    constructor: {
+      value: errorConstructor,
+      writable: true,
+      enumerable: true,
+      configurable: true,
+    },
+    name: {
+      value: name,
+      writable: true,
+      enumerable: true,
+      configurable: true,
+    },
+    toString: {
+      value: function toString(compact) {
+        return this.name + ": " + formatError(this, compact);
+      },
+      writable: true,
+      enumerable: true,
+      configurable: true,
+    },
+  });
+}
+makeErrorish(YAMLException$1, "YAMLException");
+  
 
 var exception = YAMLException$1;
 
