diff --git a/dist/lib/index.d.ts b/dist/lib/index.d.ts
index 38ab1cb3609313d3cf2115c7203cb1e3f6421cb8..59e1352f97db93ee45d29d3beb9b49f7f313c567 100644
--- a/dist/lib/index.d.ts
+++ b/dist/lib/index.d.ts
@@ -81,7 +81,7 @@ declare const provisionSmartWallet: (address: string, amount: string) => Promise
 declare const newOfferId: () => Promise<string>;
 declare const mkTemp: (template: string) => Promise<string>;
 declare const calculateWalletState: (addr: string) => Promise<"old" | "revived" | "upgraded">;
-declare const executeOffer: (address: string, offerPromise: string | Promise<string>) => Promise<void>;
+declare const executeOffer: (address: string, offerPromise: string | Promise<string>, options?: Record<string, any>) => Promise<void>;
 declare const getUser: (user: string) => Promise<string>;
 /**
  *
