diff --git a/node_modules/starshipjs/hooks.js b/node_modules/starshipjs/hooks.js
index fb3e3cf..6b6567b 100644
--- a/node_modules/starshipjs/hooks.js
+++ b/node_modules/starshipjs/hooks.js
@@ -14,7 +14,7 @@ const useRegistry = async (configFile) => {
     const registryUrl = `http://localhost:${config.registry.ports.rest}`;
     const urls = [];
     config.chains?.forEach((chain) => {
-        urls.push(`${registryUrl}/chains/${chain.name}`, `${registryUrl}/chains/${chain.name}/assets`);
+        urls.push(`${registryUrl}/chains/${chain.id}`, `${registryUrl}/chains/${chain.id}/assets`);
     });
     config.relayers?.forEach((relayer) => {
         urls.push(`${registryUrl}/ibc/${relayer.chains[0]}/${relayer.chains[1]}`, `${registryUrl}/ibc/${relayer.chains[1]}/${relayer.chains[0]}`);
@@ -35,10 +35,10 @@ const useChain = (chainName) => {
     const chainInfo = registry.getChainInfo(chainName);
     const chainID = chainInfo.chain.chain_id;
     const getRpcEndpoint = () => {
-        return `http://localhost:${config.chains.find((chain) => chain.name === chainID).ports.rpc}`;
+        return `http://localhost:${config.chains.find((chain) => chain.id === chainID).ports.rpc}`;
     };
     const getRestEndpoint = () => {
-        return `http://localhost:${config.chains.find((chain) => chain.name === chainID).ports.rest}`;
+        return `http://localhost:${config.chains.find((chain) => chain.id === chainID).ports.rest}`;
     };
     const getGenesisMnemonic = async () => {
         const url = `http://localhost:${config.registry.ports.rest}/chains/${chainID}/keys`;
@@ -50,7 +50,7 @@ const useChain = (chainName) => {
         return chainInfo.fetcher.getChainAssetList(chainName).assets[0];
     };
     const creditFromFaucet = async (address, denom = null) => {
-        const faucetEndpoint = `http://localhost:${config.chains.find((chain) => chain.name === chainID).ports.faucet}/credit`;
+        const faucetEndpoint = `http://localhost:${config.chains.find((chain) => chain.id === chainID).ports.faucet}/credit`;
         if (!denom) {
             denom = getCoin().base;
         }
